### PR TITLE
[internal][trace] Support `pf config set trace.provider` and print url for local to cloud

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

Support `pf config set trace.provider=azureml://subscriptions/<subscription_id>/resourceGroups/<resource_group>/providers/Microsoft.MachineLearningServices/workspaces/<workspace_name>` to configure local to cloud, and will record triad in `resource.attributes`:

image

If the user has enabled local to cloud feature, plus print an url for the UI link:

image

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_trace/_start_trace.py [hunk 1]
```diff
@@ -189,3 +191,32 @@ def _determine_trace_url(
     elif session_configured and session_id is not None:
         ui_url += f"?session={session_id}"
     return ui_url
+
+
+def _get_workspace_triad() -> typing.Optional[AzureMLWorkspaceTriad]:
+    # TODO: we have no conclusion on the local to cloud experience yet
+    #       one might be experience is `pf config xxx`, and we should get the triad via config;
+    #       before that, use below environ name temporarily
+    subscription_id = os.getenv("pf_test_subscription_id", None)
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_trace/_start_trace.py:200]: Let's only merge this PR when we have the pf config ready?

File: src/promptflow/promptflow/_trace/_start_trace.py [hunk 2]
```diff
@@ -189,3 +191,32 @@ def _determine_trace_url(
...
+    resource_group_name = os.getenv("pf_test_resource_group_name", None)
+    workspace_name = os.getenv("pf_test_workspace_name", None)
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_trace/_start_trace.py:202]: who sets these env vars?

File: src/promptflow/promptflow/_trace/_start_trace.py (hunk 1 above)
Comment (zhengfeiwang) [reply src/promptflow/promptflow/_trace/_start_trace.py:200]: `pf config set trace.provider=azureml:/subscriptions/<subscription_id>/resourceGroups/<resource_group>/providers/Microsoft.MachineLearningServices/workspaces/<workspace_name>`

File: src/promptflow/promptflow/_trace/_start_trace.py (hunk 2 above)
Comment (zhengfeiwang) [reply src/promptflow/promptflow/_trace/_start_trace.py:202]: offline synced, will use `pf config`

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/promptflow/_sdk/_utils.py [hunk 3]
```diff
@@ -1188,3 +1190,17 @@ def parse_otel_span_status_code(value: int) -> str:
         return "Ok"
     else:
         return "Error"
+
+
+def extract_workspace_triad_from_trace_provider(trace_provider: str) -> AzureMLWorkspaceTriad:
+    match = re.match(AZURE_WORKSPACE_REGEX_FORMAT, trace_provider)
+    if not match or len(match.groups()) != 5:
+        raise ValueError(
+            "Malformed trace provider string, expected azureml:/subscriptions/<subscription_id>/"
```
Comment (brynn-code) [inline src/promptflow/promptflow/_sdk/_utils.py:1199]: azureml://subscripts.... (also please help refine the connection one)

**Review by huaiyan — APPROVED** [review]

**Review by brynn-code — APPROVED** [review]

File: src/promptflow/promptflow/_sdk/_utils.py (hunk 3 above)
Comment (zhengfeiwang) [reply src/promptflow/promptflow/_sdk/_utils.py]: updated here and there (connection)

**Review by huaiyan — APPROVED** [review]

**Pull request merged**